### DISCUSSION TITLE
Possibility to remove a created SOC user

### DISCUSSION THREAD

**Discussion by madcaban** [Security-Onion-Solutions/securityonion#8064] [Unsupported Versions] [answered]
Is it possible to completely delete a created SOC user? I have not seen any way to do it.

Thank you and best regards

**Comment by xfaith** [comment]
in so-user looks like there might be a function for deleteUser, but doesnt look like its enabled at this time (About the only thing I think you can do at this time is disable user).

**Reply by xfaith** [reply]
So not recommended, but what I did was copy the so-user-add file to so-user-delete and inside the file change the 'add' to 'delete'.  It seems to call the deleteuser function in the so-user.  It does delete the user requested (email address required).  Unsure if there are any further complications at this time.    Of course also remember that these changes (new file) will be rremoved on update/salts pushes (i would think).

SecurityOnion folks please remove this comment as needed, I dont want folks breaking things ;).

**Reply by madcaban** [reply]
Thank you very much for your answer, xfaith. If a new binary is added as you suggested, it is possible to delete the old SOC users.

It is an unsupported solution, however, I think it would be of interest to add it in a new version.

**Comment by dougburks** [comment, marked as answer]
The recommended and supported option is to disable the user account as this prevents the user from logging in while preserving any associations with that account:
https://docs.securityonion.net/en/2.3/disabling-accounts.html

For example, if you assign a case to a user and then delete that user account, the case will show as assigned to a user ID rather than a human readable username.

If you really want to delete a user account and understand the side effects, then there is no need to create a so-user-delete script as you can simply call the deleteuser function from so-user.
